azure-pipelines/main-pipeline.yml

```diff
@@ -0,0 +1,228 @@
+    steps:
+
+# This script will check the Repo for changes and store them in a variable updatedFiles.
+# The script will iterate over every object stored in updatedFiles to select paths with changes.
+# Pipeline variable will be updated from 'false' to 'True' if a change was made in that path.
+# This step controls which build tasks will be triggered.
+# If Base is to be rebuilt, it will then set those services for which it is a dependency to build as well.   
+
+    - powershell: |
+        $updatedFiles = git diff HEAD HEAD~ --name-only
+          $updatedFiles | ForEach-Object {
+              Switch -Wildcard ($_ ) {
+                  'accounts/*' { Write-Output '##vso[task.setvariable variable=accounts;isOutput=true]True' }
+                  'base/*' { 
+                      Write-Output '##vso[task.setvariable variable=base;isOutput=true]True' 
+                      Write-Output '##vso[task.setvariable variable=accounts;isOutput=true]True'
+                      Write-Output '##vso[task.setvariable variable=commerce;isOutput=true]True'
+                      Write-Output '##vso[task.setvariable variable=inventory;isOutput=true]True'
+                  }
+                  'commerce/*' { Write-Output '##vso[task.setvariable variable=commerce;isOutput=true]True' }
+                  'config/*' { Write-Output '##vso[task.setvariable variable=config;isOutput=true]True' }
+                  'eureka-server/*' { Write-Output '##vso[task.setvariable variable=eureka-server;isOutput=true]True' }
+                  'inventory/*' { Write-Output '##vso[task.setvariable variable=inventory;isOutput=true]True' }
+                  'Spring-Cloud-Gateway/*' { Write-Output '##vso[task.setvariable variable=Spring-Cloud-Gateway;isOutput=true]True' }
+              }
+          }
+      name: ProduceVar
+
+# Printing a To-Do (Build) list. Note, that these are dynamic variables and if no variables exist
+# The script will print literal.
+
+    - script: |
+          echo 'Base will rebuild: $(ProduceVar.base)'
+          echo 'Accounts will rebuild: $(ProduceVar.accounts)'
+          echo 'Commerce will rebuild: $(ProduceVar.commerce)'
+          echo 'Inventory will rebuild: $(ProduceVar.inventory)'
+          echo 'Config will rebuild: $(ProduceVar.config)'
+          echo 'Eureka Server will rebuild: $(ProduceVar.eureka-server)'
+          echo 'Spring Cloud Gateway will rebuild: $(ProduceVar.Spring-Cloud-Gateway)'
+
+# This job ensures Base is built first, in case of dependencies.
+# Setting a variable with the output variable from the previous job.
+# It then checks that it's true, and continues on with the build, or skips.
+
+  - job: BaseBuild
+    pool:
+      name: java-16
+    dependsOn: CheckChanges
+    variables:
+        base: $[ dependencies.CheckChanges.outputs['ProduceVar.base'] ]
+    condition: eq(variables['base'], 'True')
+    steps:
+      - script: echo 'Base has been updated - $(base)'
+        displayName:  'Checking Base'
```

Comment:
isn't this step superfluous if you have the above condition?

Reply:
This has been removed.